### PR TITLE
Lets Slack detect message confirmations if optional message ID is given.

### LINKED ISSUE #31: Retrieving the reply for send_message
Hi

As noted here

https://api.slack.com/rtm

"Once the JSON has been sent to the server visual clients should immediately display the text in the channel, grayed out or otherwise marked to indicate that it is "pending". At some point after that, usually a few milliseconds later, the server will send a confirmation that the message was received:"

As it does not seem to be caught in any of the handlers, how can I retrieve this reply?

### PR DESCRIPTION
**Pull request by christhekeele**
This lets you provide an id to `send_message`, which will trigger Slack to return a confirmation when that message is posted. So it also adds a `handle_confirmation` callback so users can receive that information.

Closes #31, see discussion in #11.

### CODE REVIEW

File: lib/slack.ex [hunk 1]
```diff
@@ -46,6 +46,7 @@ defmodule Slack do
 
   * `handle_connect(slack, state)` - called when connected to Slack.
   * `handle_message(message, slack, state)` - called when a message is received.
+  * `handle_confirmation(confirmation, slack, state)` - called when a message sent with an id has been confirmed as sent by Slack.
```
Comment (BlakeWilliams) [inline lib/slack.ex]: Should wrap at 80 characters.

**Comment by CthulhuDen** [comment]
Any particular reasons for this not getting pulled? This functionality is pretty much required for a bot to be able to update it's messages (I have to use web api currently to obtain my ts'es)

**Comment by theworkerant** [comment]
@BlakeWilliams All good here?

**Comment by acconrad** [comment]
@theworkerant @christhekeele once the merge conflicts go in feel free to send another review

**Comment by theworkerant** [comment]
Wow! 2017! I don't even remember this contribution, I think my review may be of negligible value but happy to do so.

**Pull request closed without merging**